src/features/navbar/Navbar.tsx

```diff
@@ -1,12 +1,45 @@
-import {Link, Outlet} from "react-router-dom";
-import {LuShoppingCart} from "react-icons/lu";
-import {useCartStore} from "../cart/cartStore.ts";
+import { Link, Outlet } from "react-router-dom";
+import { LuHeart, LuShoppingCart } from "react-icons/lu";
+import { MdClose } from "react-icons/md";
+import { useCartStore } from "../cart/cartStore.ts";
+import { useRef } from "react";
 
 const Navbar = () => {
     const totalNumItems = useCartStore((state) => state.cartNumItems);
+    const drawerRef = useRef<HTMLInputElement>(null);
+
+    const closeDrawer = () => {
+        if (drawerRef.current) {
+            drawerRef.current.checked = false;
+        }
+    };
+
+    const NavbarLinks = () => (
```

Comment:
Consider moving this NavBarLinks component into its own .tsx file and just import it here